### PR TITLE
Adding cluster id to get cluster and get cluster list rest replies

### PR DESCRIPTION
**Pull request by alika**

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -146,10 +146,8 @@ message GetClusterMsg {
 message GetClusterReply {
     // Is the cluster in the system
     bool ok = 1;
-    // What is the status of the cluster
-    string status = 2;
-    // What is the kubeconfig to connect to the cluster
-    string kubeconfig = 3;
+    // The cluster info with kubeconfig
+    ClusterItem cluster = 2;
```
Comment (venezia) [inline api/api.proto]: What's the benefit of having a `bool, ClusterItem{string,string,string, string}` over `bool, string, string, string, string`

I'm not necessarily against it, I just want to understand the motivation.

Comment (alika) [reply api/api.proto]: The motivation being I wanted a consistent *summary* and *detailed* cluster object being returned. So I'd like this in the end:

```
{
  ok: true,
  cluster: {
    id: "xxxx-xxxx-xxxx",
    name: "managed-cluster-x",
    status: "Ready",
    kubeconfig: "apiVersion ......"
  }
}
```
for a successful, detailed cluster response and:
```
{
  ok: false,
  error: {
    code: "ECONNRESET",
    message: "socket hang up"
  }
}
```
for an unsuccessful cluster response. But I'm not be consistent on the return of the create cluster. So I think I should modify that as well.

Comment (alika) [reply api/api.proto]: After playing with the api.proto a little, this seems to go down a rabbit hole. I'm going to keep flat as you suggested.

**Comment by alika** [comment]
@venezia I've updated api.proto to use your suggested `oneof` option. Let me know what you think.

**Pull request merged**